src/main/kotlin/com/openlattice/transporter/processors/TransporterPropagateDataEntryProcessor.kt

```diff
@@ -0,0 +1,90 @@
+package com.openlattice.transporter.processors
+
+import com.hazelcast.core.Offloadable
+import com.openlattice.assembler.AssemblerConnectionManager
+import com.openlattice.assembler.AssemblerConnectionManagerDependent
+import com.openlattice.edm.EntitySet
+import com.openlattice.postgres.PostgresArrays
+import com.openlattice.rhizome.hazelcast.entryprocessors.AbstractReadOnlyRhizomeEntryProcessor
+import com.openlattice.transporter.hasModifiedData
+import com.openlattice.transporter.tableName
+import com.openlattice.transporter.types.TransporterColumnSet
+import com.openlattice.transporter.updateOneBatchForProperty
+import io.prometheus.client.Counter
+import org.slf4j.LoggerFactory
+import java.util.*
+
+class TransporterPropagateDataEntryProcessor(val entitySets: Set<EntitySet>, val partitions: Collection<Int>):
+        AbstractReadOnlyRhizomeEntryProcessor<UUID, TransporterColumnSet, Void?>(),
+        Offloadable,
+        AssemblerConnectionManagerDependent<TransporterPropagateDataEntryProcessor>
+{
+    companion object {
+        private val logger = LoggerFactory.getLogger(TransporterPropagateDataEntryProcessor::class.java)
+        val counter: Counter = Counter.build()
+                .namespace("transporter")
+                .name("rows_updated")
+                .help("Rows updated in the transporter database")
+                .register()
+    }
+
+    @Transient
+    private lateinit var acm: AssemblerConnectionManager
+
+    override fun process(entry: Map.Entry<UUID, TransporterColumnSet>): Void? {
+        transport(entry)
+        return null
+    }
+
+    private fun transport(entry: Map.Entry<UUID, TransporterColumnSet>) {
+        if (entitySets.isEmpty() || partitions.isEmpty()) {
+            return
+        }
+        val tableName = tableName(entry.key)
+        val propertyIds = entry.value.columns.keys
+        val entitySetIds = entitySets.map { it.id }.toSet()
+        val transporter = acm.connect("transporter")
+        entitySets.filter {
+            if (it.isLinking) {
+                logger.error("Skipping linking entity set {} ({})", it.name, it.id)
+            }
+            !it.isLinking
+        }
+        if (!hasModifiedData(transporter, entitySetIds, propertyIds, partitions)) {
```

Comment:
Won't this call be too big? It's querying with all the entity set ids of an entity type.
We used flags before to flag materialized entity sets, when data was updated

Reply:
the where exists (subquery) should be an extremely optimized path, but I can't know for sure without benchmarking this against prod-sized data.

Reply:
👍

Reply:
I updated this to use the ids table, which should be even faster.